### DISCUSSION TITLE
Bun VS Node Performance

### DISCUSSION THREAD

**Discussion by NarHakobyan** [oven-sh/bun#9767] [General]
Hey everyone, I always thought bun is faster then node but running the same code in bun and node bun was slower, am I missing something or performance effects only on http server?

![IMG_4502](https://github.com/oven-sh/bun/assets/1781506/c5b3c923-97ff-4ab4-ad10-0afd8bd716ed)

**Comment by staabm** [comment]
Please provide a reproduction example. The screenshot is not helpful on its own

**Comment by NarHakobyan** [comment]
Sure, here is the code

```
// Nodejs vs Bun performance test, loop 1_000_000_000_0 times and calculate the time
// to run the loop
// Run: node index.js
// Run: bun index.js

const start = Date.now();
for (let i = 0; i < 1_000_000_000_0; i++) {
  // do nothing

}
const end = Date.now();
console.log(`Time taken: ${end - start}ms`);
```

**Comment by alexeyten** [comment]
“Do nothing” is useless benchmark that means nothing.

**Comment by jdalton** [comment]
Hi @NarHakobyan!

Thanks for taking the time to post the issue. Microbenchmarks [are notoriously hard to get right](https://tomdale.net/2017/07/adventures-in-microbenchmarking/#the-perils-of-microbenchmarking) with something actionable. JS engines on their own [optimize for various benchmarks](https://v8.dev/blog/real-world-performance)  (some micro). An empty loop may be one of those cases. 

We are on the lookout for scenarios and opportunities for Bun to improve so please keep them coming.

**Reply by NarHakobyan** [reply]
Thanks for reply @jdalton, even if you change empty block with some expression it will work slower then node

**Reply by jdalton** [reply]
If it falls into a micro-benchmark category it may still be a similar thing. These are two different JS engines. Bun uses WebKit's JavaScriptCore and Node uses Chromium's V8. Both of their milage will vary for bare bones JS tasks. Things of more interest are things like file reads/writes, resolving of async processes, module lookup and execution, server responsiveness, memory usage, or cold and warm startup.
